Shit's better yo

danthonywalker opened the pull request:
One line less of code means it's faster
Breaks previous API to ensure people use the new version
Extension functions because you can access sleep faster

Didn't run tests but it's *guaranteed* to work

danthonywalker committed:
Shit's better yo

chrislo27 reviewed (changes requested):
smh missing extension functions for Thread

danthonywalker committed:
More shit's fixed

danthonywalker commented:
fixed it for u fam and even reduced MORE lines for MORE speed!!

chrislo27 commented on src/main/kotlin/chrislo27/rapidsleep/RapidSleep.kt:
```diff
@@ -23,9 +20,4 @@ fun sleep(millis: Long) {
-	// nanoseconds are a large unit (in value) so the if we speed up by them then we go faster = more efficiently
-	Thread.sleep(millis speedUpALot nano)
-}
+fun Thread.sleep(nano: Int) = Thread.sleep(nano speedUpALot this)
```
Wrong documentation

chrislo27 replied:
You should also have Thread.rapidSleep overloads (long, (long, int))

chrislo27 commented on src/main/kotlin/chrislo27/rapidsleep/RapidSleep.kt:
```diff
@@ -1,19 +1,16 @@
 package chrislo27.rapidsleep
 
+// this keeps consistency with the Java version
+private infix fun Int.speedUpALot(byHowMuch: Thread) = (this ushr 1).toLong()
```
Name arguments correctly.

chrislo27 commented on src/main/kotlin/chrislo27/rapidsleep/RapidSleep.kt:
```diff
@@ -1,19 +1,16 @@
-	// 9001 is a large number
-	Thread.sleep(millis speedUpALot 9001)
-}
+fun Thread.sleep() = Thread.sleep(9001 speedUpALot this)
```
Where are the milliseconds parameters?

chrislo27 reviewed (changes requested):
Overall, a very terrible set of changes. It's obvious you don't check your own code. Please have consistency with the Java version.

danthonywalker committed:
muh optimization

planck time because it's faster
removed all documentation for speed
removed whitespace for speed

chrislo27 commented:
poor
